pkgs/by-name/on/onedrivegui/package.nix

```diff
@@ -0,0 +1,67 @@
+{ lib
+, python3Packages
+, fetchFromGitHub
+, copyDesktopItems
+, onedrive
+, makeDesktopItem
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "onedrivegui";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "bpozdena";
+    repo = "OneDriveGUI";
+    rev = "v${version}";
+    hash = "sha256-HutziAzhIDYP8upNPieL2GNrxPBHUCVs09FFxdSqeBs=";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems ];
+
+  propagatedBuildInputs = [ onedrive ] ++ (with python3Packages; [ pyside6 requests ]);
+
+  doCheck = false; # No tests defined
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "OneDriveGUI";
+      exec = "onedrivegui";
+      desktopName = "OneDriveGUI";
+      comment = "OneDrive GUI Client";
+      type = "Application";
+      icon = "OneDriveGUI";
+      terminal = false;
+      categories = [ "Utility" ];
+    })
+  ];
+
+  preBuild = ''
+    # Patch OneDriveGUI.py so DIR_PATH poitns to shared files location
+    sed -i "s|\(DIR_PATH = \)os.path.*|\1'$out/share/OneDriveGUI'|" src/OneDriveGUI.py
```

Comment:
This allows the `OneDriveGUI.py` binary to find the asset folders in `$out/share/OneDriveGUI`